egs/wsj/s5/steps/data/reverberate_data_dir.py

```diff
@@ -0,0 +1,363 @@
+                        "--receiver-position-id <string,optional> --source-position-id <string,optional> "
+                        "--rt-60 < <float,optional> --drr <float, optional> < location(support Kaldi IO strings) >")
+    parser.add_argument("--noise-list-file", type=str, default = None,
+                        help="Noise information file, the format of the file is"
+                        "--noise-id <string,compulsary> --noise-type <choices = (isotropic, point source),compulsary> "
+                        "--bg-fg-type <choices=(background|foreground), default=background> "
+                        "--rir-file <str, compulsary if isotropic, should not be specified if point-source> "
+                        "< location=(support Kaldi IO strings) >")
+    parser.add_argument("--num-replications", type=int, dest = "num_replica", default = 1,
+                        help="Number of replicate to generated for the data")
+    parser.add_argument('--foreground-snrs', type=str, dest = "foreground_snr_string", default = '20:10:0', help='snrs for foreground noises')
+    parser.add_argument('--background-snrs', type=str, dest = "background_snr_string", default = '20:10:0', help='snrs for background noises')
+    parser.add_argument('--prefix', type=str, default = None, help='prefix for the id of the corrupted utterances')
+    parser.add_argument("--speech-rvb-probability", type=float, default = 0.8,
+                        help="Probability of reverberating the speech signal, e.g. 0 <= p <= 1")
+    parser.add_argument("--noise-adding-probability", type=float, default = 0.4,
+                        help="Probability of adding point-source noises, e.g. 0 <= p <= 1")
+    parser.add_argument("--max-noises-added", type=int, default = 2,
+                        help="Maximum number of point-source noises could be added")
+    parser.add_argument('--random-seed', type=int, default=0, help='seed to be used in the randomization of impulese and noises')
+    parser.add_argument("input_dir",
+                        help="Input data directory")
+    parser.add_argument("output_dir",
+                        help="Output data directory")
+
+    print(' '.join(sys.argv))
+
+    args = parser.parse_args()
+    args = CheckArgs(args)
+
+    return args
+
+def CheckArgs(args):
+    if not os.path.exists(args.output_dir):
+        os.makedirs(args.output_dir)
+
+    ## Check arguments.
+    if not os.path.isfile(args.rir_list_file):
+        raise Exception(args.rir_list_file + "not found")
+    
+    if args.noise_list_file is not None:
+        if not os.path.isfile(args.noise_list_file):
+            raise Exception(args.noise_list_file + "not found")
+
+    if args.num_replica > 1 and args.prefix is None:
+        args.prefix = "rvb"
+        warnings.warn("--prefix is set to 'rvb' as --num-replications is larger than 1.")
+
+    return args
+
+
+def PickItemFromDict(dict):
```

Comment:
You don't need two methods `PickItemFromDict` and `PickItemFromList`. You could check the type of the input argument inside the method and convert to list if the input is a dict.